proxy/envoy/config.go

```diff
@@ -350,7 +350,33 @@ func applyInboundAuth(listener *Listener, mesh *proxyconfig.ProxyMeshConfig) {
 }
 
 // buildTCPListener constructs a listener for the TCP proxy
-func buildTCPListener(tcpConfig *TCPRouteConfig, ip string, port int) *Listener {
+// in addition, it enables mongo proxy filter based on the protocol
+func buildTCPListener(tcpConfig *TCPRouteConfig, ip string, port int, protocol model.Protocol) *Listener {
+	// TODO: add a watcher for /var/lib/istio/mongo/certs
+	// if certs are found use, TLS or mTLS clusters for talking to MongoDB.
+	// User is responsible for mounting those certs in the pod.
+	if protocol == model.ProtocolMONGO {
+		return &Listener{
+			Name:    fmt.Sprintf("mongo_%s_%d", ip, port),
+			Address: fmt.Sprintf("tcp://%s:%d", ip, port),
+			Filters: []*NetworkFilter{{
+				Type: both,
+				Name: MONGOProxyFilter,
+				Config: &MONGOProxyFilterConfig{
+					StatPrefix: "mongo",
+				},
+			},
```

Comment:
can you share this block with the next one ? (read network filter seems to be exactly the same)

Reply:
They are different.. The one below has a different statsprefix, route config, etc. Also, mongo filter is bidirectional (both) while tcp is just read.

Reply:
the read part is the same, could be moved up in a variable and used in both blocks but it's not very important (though less lines the better most times)

Reply:
Yes, please share code for read TCP proxy filter.

Reply:
Also please switch statements for protocols, it's easier to find related pieces by protocol.